islandora_social_metatags.module

```diff
@@ -69,44 +108,27 @@ function islandora_social_metatags_islandora_view_object() {
 
   $pid = $object->id;
 
-  // Get Image path. TO DO: Error handling if no image exists.
-  if (in_array('ir:citationCModel', $object->models)) {
-        $image_datastream = variable_get('islandora_social_metatags_citation_datastream', 'PREVIEW');
-  }
-  elseif (in_array('ir:thesisCModel', $object->models)) {
-    $image_datastream = variable_get('islandora_social_metatags_thesis_datastream', 'PREVIEW');
-  }
-  elseif (in_array('islandora:sp_large_image_cmodel', $object->models)) {
-   $image_datastream = variable_get('islandora_social_metatags_largeimage_datastream', 'JPG');
-  }
-  elseif (in_array('islandora:sp_basic_image', $object->models)) {
-   $image_datastream = variable_get('islandora_social_metatags_basicimage_datastream', 'MEDIUM_SIZE');
-  }
-  elseif (in_array('islandora:sp_videoCModel', $object->models)) {
-   $image_datastream = variable_get('islandora_social_metatags_video_datastream', 'TN');
+  // Compound objects have multiple images.
+  if(in_array('islandora:compoundCModel', $object->models)) {
+    $parts = islandora_compound_object_get_parts($object->id);
+    foreach ($parts as $part) {
+      $child_object = islandora_object_load($part);
+      $child_image_datastream = _ism_get_image_datastream($child_object);
+      $images[] = $base_url . "/islandora/object/" . $child_object->id . "/datastream/" . $child_image_datastream . "/view";
+    }
   }
   else {
-    $image_datastream = "TN";
+    // Get Image path. TO DO: Error handling if no image exists.
+    $image_datastream = _ism_get_image_datastream($object);
+    $images[] = $base_url . "/islandora/object/" . $pid . "/datastream/" . $image_datastream . "/view";
   }
 
-  global $base_url;
-  $image = $base_url . "/islandora/object/" . $pid . "/datastream/" . $image_datastream . "/view";
   $og_url = $base_url . "/islandora/object/" . $pid;
-// Check whether the image datastream actually exists..
 
-  $file = $image;
-  $file_headers = @get_headers($file);
 
-  if (!$file_headers || strpos($file_headers[0], '404')) {
-    $exists = FALSE;
-  }
-  else {
-    $exists = TRUE;
-  }
- 
 // If the datastream does not exist, use the default image path. 
-  if (!$exists) {
-    $image = $base_url . variable_get('islandora_social_metatags_noimage', '/sites/all/modules/islandora/images/folder.png');
+  if (empty($images)) {
+    $images[] = $base_url . variable_get('islandora_social_metatags_noimage', '/sites/all/modules/islandora/images/folder.png');
   }
```

Comment:
This doesn't work... if datastream doesn't exist, it is still printing a link to it in the meta tag.

Reply:
Hi Brandon. I assumed multiple images would be displayed appropriately, but I hadn't actually tested (I don't have a world-accessible site that I've tested this on yet)! 

Your ideas for selecting which preview image from compound objects to use are helpful. I'll give that some more thought. But I wonder if it maybe makes most sense to just use the first child, rather than try to second-guess.

I'll fix the missing image bug also, 

I'll let you know when I have an updated pull request.  Thanks!